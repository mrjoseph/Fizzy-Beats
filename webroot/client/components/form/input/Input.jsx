import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Label, Span, InputStyled, UploadInput, UploadLabel } from './Input.styles';
import './input.css';

class Input extends Component {
  errorClass() {
    const { formErrors, name } = this.props;
    if (formErrors[name]) {
      return (formErrors[name].valid) ? '#abe2b7' : 'red';
    }
    return '#fff';
  }
  handleDrag = () => {
    console.log('do something')
  }
  handleLeave = () => {
    console.log('do something else')
  }
  render() {
    const {
      name, text, onChange, type, handleBlur, multiple, accept
    } = this.props;
    
    return (
      <div className="field">
       {type === 'file' ?
        <UploadLabel htmlFor={name}>
          <UploadInput 
            theme={{ error: this.errorClass() }}
            type={type}
            onChange={onChange}
            onBlur={handleBlur}
            onDragEnter={this.handleDrag}
            onDragLeave={this.handleLeave}
            name={name}
            id={name}
            placeholder={name}
            className="input"
            multiple={multiple}
            accept={accept}
          /> 
          <div>{text}</div>
       </UploadLabel>
       : 
       <Label htmlFor={name}>
        <InputStyled
          theme={{ error: this.errorClass() }}
          type={type}
          onChange={onChange}
          onBlur={handleBlur}
          name={name}
          id={name}
          placeholder={name}
          className="input"
        /> 
        <Span>{text}</Span>
      </Label>
      }  
        {/* {this.errorWarnings()} */}
      </div>
    );
  }
}

Input.propTypes = {
  onChange: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
  formErrors: PropTypes.shape({
    valid: PropTypes.bool,
  }).isRequired,
  text: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};
Input.displayName = 'Input';
export default Input;
