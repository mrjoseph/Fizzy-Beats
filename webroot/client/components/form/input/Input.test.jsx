import { shallow } from 'enzyme';
import React from 'react';
import Input from './Input';
import 'jest-styled-components';

describe('Input', () => {
  const props = {
    formErrors: {
      username: {
        field: 'username',
        message: 'empty username',
        valid: false,
      },
    },
    name: 'username',
    text: '',
    onChange: jest.fn(),
    type: '',
    id: 'username',
    handleBlur: jest.fn(),
  };
  let component;


  describe('With no errors', () => {
    beforeEach(() => {
      component = shallow(<Input {...props} />);
    });

    it('should have no error class', () => {
      const input = component.find('.input');
      input.simulate('change', {
        target: {
          name: 'username',
          value: 'trevor',
        },
      });

      expect(props.onChange).toHaveBeenCalled();
      expect(props.onChange).toHaveBeenCalledWith({
        target: {
          name: 'username',
          value: 'trevor',
        },
      });
      expect(input.hasClass('error')).toBe(true);
    });
  });

  describe('With error', () => {
    const formErrors = {
      username: {
        field: 'username',
        message: '',
        valid: true,
      },
    };
    const propsWithError = {
      formErrors, ...props,
    };
    beforeEach(() => {
      component = shallow(<Input {...propsWithError} />);
    });
    it('input should have a error class', () => {
      const input = component.find('.input');
      expect(input.hasClass('error')).toBe(true);
    });
  });

  describe('With success', () => {
    const formErrors = {
      username: {
        field: 'username',
        message: '',
        valid: false,
      },
    };
    const propsWithError = {
      formErrors, ...props,
    };
    beforeEach(() => {
      component = shallow(<Input {...propsWithError} />);
    });
    it('input should have a success class', () => {
      const input = component.find('.input');
      expect(input.hasClass('success')).toBe(false);
    });
  });
});
