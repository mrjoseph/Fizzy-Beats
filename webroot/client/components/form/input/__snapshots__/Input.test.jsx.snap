// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input InputStyled styled component renders correctly 1`] = `
.c0 {
  border: 1px solid;
  font: 18px Helvetica,Arial,sans-serif;
  box-sizing: border-box;
  display: block;
  padding: 18px;
  width: 300px;
  margin-bottom: 20px;
  font-size: 18px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0::-webkit-input-placeholder {
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  color: #999;
  font: 18px Helvetica,Arial,sans-serif;
}

.c0::-moz-placeholder {
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  color: #999;
  font: 18px Helvetica,Arial,sans-serif;
}

.c0:-ms-input-placeholder {
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  color: #999;
  font: 18px Helvetica,Arial,sans-serif;
}

.c0::placeholder {
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  color: #999;
  font: 18px Helvetica,Arial,sans-serif;
}

.c0:focus + span {
  opacity: 1;
  top: 10px;
}

.c0:focus,
.c0.populated {
  padding-top: 26px;
  padding-bottom: 10px;
}

.c0:focus::-webkit-input-placeholder,
.c0.populated::-webkit-input-placeholder {
  color: transparent;
}

.c0:focus::-moz-placeholder,
.c0.populated::-moz-placeholder {
  color: transparent;
}

.c0:focus:-ms-input-placeholder,
.c0.populated:-ms-input-placeholder {
  color: transparent;
}

.c0:focus::placeholder,
.c0.populated::placeholder {
  color: transparent;
}

<input
  className="c0"
/>
`;

exports[`Input UploadInput styled component renders correctly 1`] = `
.c0 {
  border: 1px solid;
}

<input
  className="c0"
/>
`;
